src/docs/sphinx/quickstart.rst

```diff
@@ -267,28 +267,268 @@ must be specified using either:
+* cmake
+* autoconf
+* automake
+* graphviz
+* pkgconfig
+* xorg-libX11
 
-Install required compilers and MPI via `homebrew <https://brew.sh/>`_:
+Activate the particular package versions with MacPorts:
 
 .. code-block:: bash
 
-  $ brew install gcc@8
-  $ brew install llvm@11
-  $ brew install mpich
+   $ sudo port select clang mp-clang-12
+   $ sudo port select mpi openmpi-clang-fortran
+   $ sudo port select gcc mp-gcc12
 
-Build third-party libraries via Uberenv:
+This step tells MacPorts to make symbolic links in your path so that, for example, ``clang`` will invoke the clang12 from the MacPorts package and not the ``clang`` shipped by Apple. 
+It also sets up a set of symlinks so that Clang, GCC, and the MPI wrappers all work without you having to muck with environment variables to locate header files and libraries. 
+It may be possible to skip this step and give full paths to your compilers in ``compilers.yaml`` (instead of the symlinks ``/opt/local/bin/clang``, etc.), but we haven't tried this.
 
-``python3 scripts/uberenv/uberenv.py --spec=%apple-clang@11.0.1 --prefix=../path/to/install``
+While building ParMetis, spack invokes the MPI compiler wrapper with the ``mpic++`` command, but MacPorts does not create this particular synonym for OpenMPI. 
+It does create ``mpicxx``. This can be solved by making a symlink:
 
-.. note::
-   You may need to alter the compiler spec inside ``scripts/spack/configs/darwin/compilers.yaml``.
-   The checked-in version worked for one developer but often needs to be changed for installed compiler
-   paths as well as `operating_system` to match your machine.
+.. code-block:: bash
+
+   $ cd /opt/local/bin
+   $ sudo ln -s mpicxx mpic++
+
+MacPorts will automatically update the ``mpicxx`` symlink in ``/opt/local/bin`` to point to the correct executable when you use the ``port select`` command to activate a partiular MPI package. 
+By making ``mpic++`` point to ``mpicxx``, this command will also automatically point to the correct executable if you change the global MPI package through MacPorts in the future.
+
+Python environment
+^^^^^^^^^^^^^^^^^^
+
+The Serac build scripts install the ``clingo`` package in your Python environment (and may even *uninstall* it if it finds it with a version it considers too old). 
+I don't like to let software packages change my Python environment, particularly by adding or removing packages with ``pip``, as this can lead to version dependency conflicts.  
+I create a Python virtual environment in the serac root directory with the built-in ``venv`` module. 
+
+.. (I use the name "venv" for the virtual environment; We might consider adding `venv/` to the list of git ignores in the root directory).
+
+.. code-block:: bash
+
+   $ python3 -m venv <virtual-env-name>
+
+Since spack will need ``clingo``, activate the python vitual environment and 
+install the package to meet this dependency:
+
+.. code-block:: bash
+
+   $ source <virtual-env-name>/bin/activate
```

Comment:
I would skip this activation step and tell users to run `<virtual-env-name>/bin/pip3 install clingo` to avoid having to worry about the state of environment variables.

Similarly, when invoking the `uberenv.py` script, I'd tell users to run `<virtual-env-name/bin/python3 ./scripts/uberenv/uberenv.py --prefix=../path/for/TPLs` etc., to avoid the environment variables state issues.

Unexpected environment state is a really common failure mode, which is why many package managers go out of their way to sandbox environment variables; it will save you a lot of headaches if you sidestep this issue entirely by advising users to use absolute paths instead of activating an environment.

Reply:
I can see why setting the environment state could be dangerous. But in this case, doesn't the virtual environment make the user environment *more* predictable? The user could have any setting for `PYTHONHOME` or `PYTHONPATH`, and those could change the behavior of `<virtual-env-name>/bin/pip3`. The virtual environment will clear the python environment variables and make sure that pip looks for and installs packages where we expect.

Reply:
> But in this case, doesn't the virtual environment make the user environment _more_ predictable?

It doesn't if they've already activated a virtual environment instantiated via `venv` or `virtualenv`, or some other software (e.g., `conda` from Anaconda's Python distro or `lmod`) is managing environment variables. "Other software managing the environment variables" tends to be the failure mode I saw at LLNL.

> The user could have any setting for `PYTHONHOME` or `PYTHONPATH`, and those could change the behavior of `<virtual-env-name>/bin/pip3`.

That's true. But in this scenario, those environment variables would also affect the instantiation of the virtual environment because one could manipulate those environment variables so that alternate versions of `venv` and `pip` take precedence over the "stock" versions. That pathology could be mitigated by instantiating the virtual environment via:

```console
% /absolute/path/to/python3 -E -m venv <virtual-env-name>
```
and provisioning the virtual environment via:

```console
% /absolute/path/to/<virtual-env-name>/bin/python3 -E -m pip install clingo
```

or, if you want, you could write a wrapper script that unsets `PYTHONHOME` and `PYTHONPATH` in that scope, or in a subshell. Python's `-s` and `-I` flags could also be used on sufficiently recent Pythons to provide further isolation, if that's what you're going for.

I usually saw `PYTHONPATH` used in DTED environments to kludge around incorrect `RPATH`s in shared libraries (e.g., due to a borked Silo install, or Cubit's odd `RPATH`-ing, at least prior to version 15.9). Typically, this environment variable didn't materially affect building something like `clingo` because the only paths in `PYTHONPATH` pointed to Silo libraries and maybe Cubit libraries. But again, as you correctly point out, it is possible to set that variable and/or `PYTHONHOME` in ways that will foul up script execution.

Reply:
It's true, there are really an infinite number of ways that a user's environment could be surprising. This is convincing me to say less. I'm just going to say that the script will try to install the clingo package if it's not already present. The user can decide if and how they want to isolate that.